### PR TITLE
Feature: Layout Builder default content

### LINKED ISSUE #1925: Layout Builder default content
# Problem
To be able to fully switch over to Layout Builder, we need to be able to replace the default homepage provided in SiteNow with something built in Layout Builder.

The Default Content module currently doesn't provide this functionality, as it depends on the built-in REST API and that doesn't support exporting layouts or layout blocks yet.

# Proposed solution
There is an issue to add support for Layout Builder-related content via REST and JSON: https://www.drupal.org/project/drupal/issues/2942975

1. Apply the latest patch to see if it works: https://www.drupal.org/files/issues/2020-07-07/2942975-116.patch
2. Using the SiteNow demo site (default), re-create the content using Layout Builder
3. Establish menu links for view pages (news, people) from #1081 ?
4. Contact form from #1415 ?
5. Re-export the content
6. ???
7. Profit!

### PR DESCRIPTION
**Pull request by pyrello**
Resolves: #1925 

See this comment for what needs to be done to finish the work on this: https://github.com/uiowa/uiowa/blob/1277ebbc13d305a1cfea42a7c7c937e14f249348/docroot/profiles/custom/sitenow/content/block_content/0c0c1f36-3804-48b0-b384-6284eed8c67e.json#L12

# To test
* blt drupal:install --site=cogscilang.grad.uiowa.edu
* Confirm that the default content installs and that there are no visible issues with it

### CODE REVIEW

**Comment by pyrello** [comment]
This is blocked by #2032.

**Comment by pyrello** [comment]
This is also blocked by #2071

**Comment by pyrello** [comment]
@bspeare When you have a chance, I don't seem to be able to get the spacing right on the Getting started section. I could use some help making sure that I'm not just selecting the wrong combination of options.

File: docroot/profiles/custom/sitenow/content/block_content/0c0c1f36-3804-48b0-b384-6284eed8c67e.json
Comment (joewhitsitt) [inline docroot/profiles/custom/sitenow/content/block_content/0c0c1f36-3804-48b0-b384-6284eed8c67e.json]: My OCD thanks you!

Comment (pyrello) [reply docroot/profiles/custom/sitenow/content/block_content/0c0c1f36-3804-48b0-b384-6284eed8c67e.json]: I would like to fix the revision numbers too, but I'm not sure if revision IDs are unique per entity type or shared across all entitities.

**Comment by pyrello** [comment]
This is now blocked by #2076

**Comment by joewhitsitt** [comment]
Can we update node/1 node/2 to start as filtered_html body? I think the minimal as default snuck into this content. new content body starts off with filtered_html

**Comment by pyrello** [comment]
# How to finish this PR
* Default content doesn't detect blocks via LB and therefore the command to export a single entity with references doesn't work for this now.
* To get around this, all components are explicitly defined here: https://github.com/uiowa/uiowa/blob/feature_lb_default_content/docroot/profiles/custom/sitenow/sitenow.info.yml. Everything defined in that file will be exported via `drush @default.local dcem`
* To get new block UUID's, you can query them via `drush @default.local sqlq "SELECT uuid FROM block_content". I believe that every block that is defined in the system should be one that is necessary.
*I was attempting to reset the IDs that get defined in the system so that they start at 1: https://github.com/uiowa/uiowa/blob/1277ebbc13d305a1cfea42a7c7c937e14f249348/docroot/profiles/custom/sitenow/content/block_content/0c0c1f36-3804-48b0-b384-6284eed8c67e.json#L12 - this isn't strictly necessary, but I thought it was good to not have their IDs start at random arbitrarily high values.

**Comment by ABoston88** [comment]
Some remaining styling stuff (and technical issues):

-Slider block: Slides are not editable after initial creation of the slider. Only option is "Duplicate" 
-Configure link doesn't work for sections 1 and 2
-Articles: Top and bottom padding for Article block is outsized in two-column layout, not changeable in LB settings

**Comment by joewhitsitt** [comment]
>-Slider block: Slides are not editable after initial creation of the slider. Only option is "Duplicate"

This will be fixed in https://github.com/uiowa/uiowa/pull/2265

**Comment by joewhitsitt** [comment]
>-Configure link doesn't work for sections 1 and 2

Homepage on default.dev

@bspeare or @cory-skeers  do you know why this is for webmasters? I can't seem to pin it down and know that there were other restrictions put in place lately.

**Comment by joewhitsitt** [comment]
>-Articles: Top and bottom padding for Article block is outsized in two-column layout, not changeable in LB settings

@ABoston88 There was an option in section settings to vertical align start. Is this what you meant?

Screen Shot 2020-09-30 at 2 33 32 PM

**Comment by ABoston88** [comment]
@joewhitsitt Hmmm...I tried that setting previously and it didn't have any effect. Glad it's working now. Will keep an eye out for it coming up again.

**Comment by cory-skeers** [comment]
> > -Configure link doesn't work for sections 1 and 2
> 
> Homepage on default.dev
> 
> @bspeare or @cory-skeers do you know why this is for webmasters? I can't seem to pin it down and know that there were other restrictions put in place lately.

Are there any console errors when trying to use the link? Wondering if it's the access dependency being wonky. I'll take a look.

Edit:
`The website encountered an unexpected error. Please try again later.` on the POST that should open the off-canvas container. Not the access dependency error I'd been thinking it might have been.

**Comment by richardbporter** [comment]
> -Configure link doesn't work for sections 1 and 2

That is resolved by #2270.

**Comment by joewhitsitt** [comment]
Alex (content) and Ben (design) both gave a thumbs up on moving forward

**Comment by joewhitsitt** [comment]
Was getting export errors following the directions related to layout builder. Noticed there was a new release for default_content. Exports to yaml now and a bunch of other things. One bad, the homepage is no longer node/1 and is instead node/2.. :(

Anyway, took the opportunity to clean up files and remove the orphans. I think the embed icon is still needed if website switches to v2 so I left that for now.

Body on home/about can't be set to filtered_html default without a content value.

**Comment by pyrello** [comment]
Pulling down for testing.

**Comment by pyrello** [comment]
I'm going through the content and cleaning it up a little bit. There are some extra sections with no content. Additionally, it looks like some of the media items are no longer in use. I'm going to remove those, unless someone thinks there is a reason to keep them.

**Comment by pyrello** [comment]
@joewhitsitt I can't mark this as approved since I created it. I have tested and think it is good to go.

**Review by joewhitsitt — APPROVED** [review]
Tested the changes. Not noticing any thing out of place. Not sure I should approve, but I approve.

Going to have to revisit once https://github.com/uiowa/uiowa/pull/2152 is in place from a feature stand point. What about when https://github.com/uiowa/uiowa/pull/2075 is merged in?

**Comment by pyrello** [comment]
@joewhitsitt With regard to #2075, I think after this is merged, I can make any necessary updates to the content in that PR. Either ditto that for #2152 or done in a follow-up issue.

**Pull request merged**
